mesheryctl/internal/cli/root/system/component.go

```diff
@@ -0,0 +1,489 @@
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return errors.New(utils.SystemModelSubError("this command takes no arguments\n", "list"))
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		var url string
+		if cmd.Flags().Changed("page") {
+			url = fmt.Sprintf("%s/api/meshmodels/components?page=%d", baseUrl, pageNumberFlag)
+		} else {
+			url = fmt.Sprintf("%s/api/meshmodels/components?pagesize=all", baseUrl)
+		}
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		componentsResponse := &models.MeshmodelComponentsAPIResponse{}
+		err = json.Unmarshal(data, componentsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Model", "kind", "Version"}
+		rows := [][]string{}
+
+		for _, component := range componentsResponse.Components {
+			if len(component.DisplayName) > 0 {
+				rows = append(rows, []string{component.Model.Name, component.Kind, component.APIVersion})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no component is found
+			fmt.Println("No components(s) found")
+			return nil
+		}
+
+		if cmd.Flag("count").Value.String() == "true" {
+			fmt.Println("Total components: ", componentsResponse.Count)
+			return nil
+		}
+
+		if cmd.Flags().Changed("page") {
+			utils.PrintToTable(header, rows)
+		} else {
+			startIndex := 0
+			endIndex := min(len(rows), startIndex+maxRowsPerPage)
+			whiteBoardPrinter := color.New(color.FgHiBlack, color.BgWhite, color.Bold)
+			for {
+				// Clear the entire terminal screen
+				utils.ClearLine()
+
+				// Print number of models and current page number
+				whiteBoardPrinter.Print("Total number of models: ", len(rows))
+				fmt.Println()
+				whiteBoardPrinter.Print("Page: ", startIndex/maxRowsPerPage+1)
+				fmt.Println()
+
+				utils.PrintToTable(header, rows[startIndex:endIndex])
+				keysEvents, err := keyboard.GetKeys(10)
+				if err != nil {
+					return err
+				}
+
+				defer func() {
+					_ = keyboard.Close()
+				}()
+
+				event := <-keysEvents
+				if event.Err != nil {
+					utils.Log.Error(fmt.Errorf("unable to capture keyboard events"))
```

Comment:
```suggestion
					utils.Log.Error(fmt.Errorf("unable to capture keyboard events"))
```
Please use the `meshkit` error.
If `meshkit logger` is used to log non-meshkit error it results in panic.